### PR TITLE
New endpoint to create a mentorship request

### LINKED ISSUE #140: New endpoint to save the application
## Description
After the mentee fill out the application form, we need to save this data.

Request:
```
POST /mentorships/${mentor.id}/apply
```

Payload:
```
{
  goals: [
    "Goal 1",
    "Goal 2"
  ],
  background: string,
  expectations: string,
  message: string,
}
```

Response:
```
{ success: true }
```

Required fields:
- `message`

AC
- Make sure there are no previous mentorship requests for this mentor and mentee
- Make sure the mentor is available
- Create the new mentorship request, set the status as `new` (Application Status available:  `new`, `viewed`, `rejected`, `accepted`, `terminated`)
- Send an email to the mentor
- Return validation errors
- Return success call

### PR DESCRIPTION
**Pull request by crysfel**
closes #140

### CODE REVIEW

**Review by chochstrasser — APPROVED** [review]
nice work

File: src/modules/common/mentors.service.ts [hunk 1]
```diff
@@ -15,13 +16,25 @@ export class MentorsService {
     private readonly applicationModel: Model<Application>,
   ) {}
 
+  /**
+   * Finds a mentor by ID
+   * @param _id
+   */
+  async findById(_id: string): Promise<User> {
+    if (isObjectId(_id)) {
+      return await this.userModel.findOne({ _id, roles: Role.MENTOR }).exec();
```
Comment (Mooli88) [inline src/modules/common/mentors.service.ts:25]: Not sure I understand the use of `async` `await` here.
```suggestion
  findById(_id: string): Promise<User> {
    if (isObjectId(_id)) {
      return this.userModel.findOne({ _id, roles: Role.MENTOR }).exec();
```

Comment (crysfel) [reply src/modules/common/mentors.service.ts:25]: There's no need for that await in there, will remove it!

File: src/modules/common/mentors.service.ts [hunk 2]
```diff
@@ -15,13 +16,25 @@ export class MentorsService {
...
+  }
+
   /**
    * Search for mentors by the given filters
    * @param filters filters to apply
    */
   async findAll(filters: MentorFiltersDto, isLoggedIn: boolean): Promise<any> {
     const onlyMentors: any = {
-      roles: 'Mentor',
+      roles: Role.MENTOR,
```
Comment (moshfeu) [inline src/modules/common/mentors.service.ts:37]: 👍

File: src/modules/mentorships/interfaces/mentorship.interface.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import { Document, ObjectID } from 'mongoose';
+
+export enum Status {
+  NEW = 'New',
+  VIEWED = 'Viewed',
+  APPROVED = 'Approved',
+  REJECTED = 'Rejected',
+  TERMINATED = 'Terminated',
```
Comment (moshfeu) [inline src/modules/mentorships/interfaces/mentorship.interface.ts:8]: I believe we should add `canceled` if the mentee already found a mentor or changed their mind.

File: src/modules/mentorships/mentorships.controller.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    if (!mentor.available) {
+      throw new BadRequestException('Mentor is not available');
+    }
+
+    const mentorship: Mentorship = await this.mentorshipsService.findMentorship(
+      mentor._id,
+      current._id,
+    );
+    if (mentorship) {
+      throw new BadRequestException('A mentorship request already exist.');
```
Comment (moshfeu) [inline src/modules/mentorships/mentorships.controller.ts:65]:
```suggestion
      throw new BadRequestException('A mentorship request already exists');
```

**Review by moshfeu — COMMENTED** [review]
I like to read your code. It's so clean and readable :)

**Pull request merged**